stan/math/prim/meta/holder.hpp

```diff
@@ -321,21 +339,39 @@ auto make_holder_impl(const F& func, std::index_sequence<Is...>,
  * This is similar to calling the functor with given arguments. Except that any
  * rvalue argument will be moved to heap first. The arguments moved to heap are
  * deleted once the expression is destructed.
+ *
  * @tparam F type of the functor
  * @tparam Args types of the arguments
  * @param func the functor
  * @param args arguments for the functor
  * @return `holder` referencing expression constructed by given functor
  */
 template <typename F, typename... Args,
-          require_eigen_t<
+          require_not_plain_type_t<
               decltype(std::declval<F>()(std::declval<Args&>()...))>* = nullptr>
 auto make_holder(const F& func, Args&&... args) {
   return internal::make_holder_impl(func,
                                     std::make_index_sequence<sizeof...(Args)>(),
                                     std::forward<Args>(args)...);
 }
 
+/**
+ * Calls given function with given arguments. No `holder` is necessary if the
+ * function is not returning Eigen expression.
+ *
+ * @tparam F type of the functor
+ * @tparam Args types of the arguments
+ * @param func the functor
+ * @param args arguments for the functor
+ * @return `holder` referencing expression constructed by given functor
+ */
+template <typename F, typename... Args,
+          require_plain_type_t<
+              decltype(std::declval<F>()(std::declval<Args&>()...))>* = nullptr>
+auto make_holder(const F& func, Args&&... args) {
+  return func(std::forward<Args>(args)...);
+}
```

Comment:
I'll take another pass of this file specifically after you grade my holder knowledge.

Reply:
You pass, so so you can make another pass.

Reply:
Dang that's good wordplay lol